lib/remote_persistent_term/fetcher/http.ex

```diff
@@ -0,0 +1,60 @@
+defmodule RemotePersistentTerm.Fetcher.Http do
+  @moduledoc """
+  A Fetcher implementation for HTTP.
+  """
+  require Logger
+
+  @behaviour RemotePersistentTerm.Fetcher
+
+  @type t :: %__MODULE__{
+          url: String.t()
+        }
+
+  defstruct [:url]
+
+  @opts_schema [
+    url: [
+      type: :string,
+      required: true,
+      doc: "The url from which the remote term is downloaded."
+    ]
+  ]
+
+  @doc """
+  Initialize a HTTP Fetcher.
+
+  Options:
+  #{NimbleOptions.docs(@opts_schema)}
+  """
+  @impl true
+  def init(opts) do
+    with {:ok, valid_opts} <- NimbleOptions.validate(opts, @opts_schema) do
+      {:ok,
+       %__MODULE__{
+         url: valid_opts[:url]
+       }}
+    end
+  end
+
+  @impl true
+  def current_version(_state) do
+    {:ok, DateTime.utc_now() |> DateTime.to_string()}
+  end
+
+  @impl true
+  def download(state) do
+    Logger.info("downloading remote term from #{state.url}")
+
+    with {:ok, resp} <- Req.get(state.url, cache: true) do
+      if resp.status < 300 do
```

Comment:
304 is the status code for "Not Modified", are you sure we want to return an error in that case?

Reply:
We wouldn't want to replace the term with the contents of that 304 body, which would be null. I would assume that Req caching would handle that for us and return a 200. I'll test that locally to see 

https://github.com/wojtekmach/req/blob/50c7d3ee2ceea2c64f88526efb49a81ec173cd9c/lib/req/steps.ex#L557

Reply:
yes req returns a 200 from a 304

Reply:
ah cool, lets add a test case that shows that. We can use [bypass](https://github.com/PSPDFKit-labs/bypass) for example

Reply:
cool, good idea